account_invoice_ubl/models/account_invoice.py

```diff
@@ -0,0 +1,328 @@
+# © 2016-2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import base64
+from lxml import etree
+import logging
+
+from odoo import models, api
+from odoo.tools import float_is_zero, float_round
+
+logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+    _name = 'account.invoice'
+    _inherit = ['account.invoice', 'base.ubl']
+
+    @api.multi
+    def _ubl_add_header(self, parent_node, ns, version='2.1'):
+        ubl_version = etree.SubElement(
+            parent_node, ns['cbc'] + 'UBLVersionID')
+        ubl_version.text = version
+        doc_id = etree.SubElement(parent_node, ns['cbc'] + 'ID')
+        doc_id.text = self.number
+        issue_date = etree.SubElement(parent_node, ns['cbc'] + 'IssueDate')
+        issue_date.text = self.date_invoice
+        type_code = etree.SubElement(
+            parent_node, ns['cbc'] + 'InvoiceTypeCode')
+        if self.type == 'out_invoice':
+            type_code.text = '380'
+        elif self.type == 'out_refund':
+            type_code.text = '381'
+        if self.comment:
+            note = etree.SubElement(parent_node, ns['cbc'] + 'Note')
+            note.text = self.comment
+        doc_currency = etree.SubElement(
+            parent_node, ns['cbc'] + 'DocumentCurrencyCode')
+        doc_currency.text = self.currency_id.name
+
+    @api.multi
+    def _ubl_add_order_reference(self, parent_node, ns, version='2.1'):
+        self.ensure_one()
+        if self.name:
+            order_ref = etree.SubElement(
+                parent_node, ns['cac'] + 'OrderReference')
+            order_ref_id = etree.SubElement(
+                order_ref, ns['cbc'] + 'ID')
+            order_ref_id.text = self.name
+
+    @api.multi
+    def _ubl_get_contract_document_reference_dict(self):
```

Comment:
I don't understand why this method. One record is ensured in parent method, and this do nothing.

Reply:
I don't understand either :thinking:

Reply:
@cubells is it ok for you to leave this method as it is and move forward?